finished readme, left off find_or_create_by for separate reading

SophieDeBenedetto opened the pull request:
needs :eyes:

@jmburges @AnnJohn

SophieDeBenedetto committed:
finished readme, left off find_or_create_by for separate reading

SophieDeBenedetto committed:
Update README.md

jmburges commented on README.md:
````diff
@@ -1,3 +1,284 @@
+# => "The Black Album"
+```
+
+Much better. Now we need to save this record back into the database. To do so, we'll need to use an `UPDATE` SQL statement. That statement would look something like this:
````
So you wouldn't want to run a SQL statement with the `WHERE` clause against the name. There may be _many_ songs with the name "99 problems". The only unique identifier for that row is the ID. This example needs to be against an id

jmburges replied:
should have kept reading JK <3

jmburges commented on README.md:
````diff
@@ -1,3 +1,284 @@
+Let's put it all together using our SQLite3-Ruby gem magic. Remember, in this example, we assume our database connection is stored in `DB[:conn]`. 
+
+```ruby
+sql = "UPDATE songs SET album=#{ninety_nine_problems.album" WHERE name = ?"
````
make it id based, not name based

jmburges replied:
should have kept reading. JK

jmburges commented on README.md:
```diff
@@ -1,3 +1,284 @@
+
+If that sounds confusing, check out this diagram:
+
+![](https://docs.google.com/drawings/d/1KYjce3m-ItD3j0hTuorBtlTgPoQSVg9v2sddGLVle8M/pub?w=958&h=497)
```
put this image on S3

SophieDeBenedetto replied:
@jmburges images on s3 not showing up. not sure why. It is in the s3 bucket, permission are "read: world".

jmburges commented:
few issues and :shipit:

SophieDeBenedetto committed:
Update README.md

SophieDeBenedetto committed:
fix s3 image 

change file extension from pdg to png

SophieDeBenedetto committed:
fix save method to avoid duplication

save method should check to see if object has been persisted. If so, simply update it using new update method we just created. Otherwise, INSERT a new record into the DB

SophieDeBenedetto committed:
fixed #save method

SophieDeBenedetto committed:
fixed #save method and all code snippets that use it

#save method should do the id assignment. some code snippets that show #save where missing that line

SophieDeBenedetto committed:
fix #create method so that it returns the object it created

SophieDeBenedetto committed:
fix picture

SophieDeBenedetto committed:
add .learn, contributing, license

SophieDeBenedetto committed:
Merge branch 'wip-new' of https://github.com/learn-co-curriculum/orm-updating-records into wip-new
